src/Workspaces/MSBuild/Core/MSBuild/IBinLogPathProvider.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.MSBuild;
+
+internal interface IBinLogPathProvider
+{
+    /// <summary>
+    /// Returns a new log path. Each call will return a new name, so that way we don't have collisions if multiple processes are writing to different logs.
```

Comment:
It would be good to mention that a null return indicates the caller is "disabling" binlogs.

Reply:
Ah, right!